src/Webhook/WebhookService.php

```diff
@@ -96,12 +106,18 @@ public function registerWebhook(?string $salesChannelId): string
             $webhookExecuteToken = Random::getAlphanumericString(self::PAYPAL_WEBHOOK_TOKEN_LENGTH);
         }
 
-        $this->router->getContext()->setScheme('https');
-        $webhookUrl = $this->router->generate(
-            'api.action.paypal.webhook.execute',
-            [self::PAYPAL_WEBHOOK_TOKEN_NAME => $webhookExecuteToken],
-            UrlGeneratorInterface::ABSOLUTE_URL
-        );
+        if ($this->proxyWebhookUrl !== null) {
+            $webhookUrl = $this->proxyWebhookUrl . '?' . http_build_query(
```

Comment:
What if the proxy already contains GET parameters? Maybe do a `str_contains('?') ? '&' : '?'` (overly simplified)